x-pack/plugins/alerting/server/alert/create_alert_factory.ts

```diff
@@ -25,28 +25,49 @@ export interface CreateAlertFactoryOpts<
 > {
   alerts: Record<string, Alert<State, Context>>;
   logger: Logger;
+  maxAlerts: number;
   canSetRecoveryContext?: boolean;
 }
 
 export function createAlertFactory<
   State extends AlertInstanceState,
   Context extends AlertInstanceContext,
   ActionGroupIds extends string
->({ alerts, logger, canSetRecoveryContext = false }: CreateAlertFactoryOpts<State, Context>) {
+>({
+  alerts,
+  logger,
+  maxAlerts,
+  canSetRecoveryContext = false,
+}: CreateAlertFactoryOpts<State, Context>) {
   // Keep track of which alerts we started with so we can determine which have recovered
   const originalAlerts = cloneDeep(alerts);
+
+  // Number of alerts reported
+  let numAlertsCreated = 0;
+
+  // Whether the number of alerts reported has reached max allowed
+  let hasReachedAlertLimit = false;
+
   let isDone = false;
   return {
     create: (id: string): PublicAlert<State, Context, ActionGroupIds> => {
       if (isDone) {
         throw new Error(`Can't create new alerts after calling done() in AlertsFactory.`);
       }
+
+      if (numAlertsCreated++ >= maxAlerts) {
+        logger.warn(`Rule run generated greater than ${maxAlerts} alerts.`);
```

Comment:
**nit:** should we include the rule type id, id, and space in the message?

Reply:
Updated in https://github.com/elastic/kibana/pull/138446/commits/24da6da3d40419c557ddd431bf0f38ac641659e2